hathor/transaction/scripts/execute.py

```diff
@@ -13,16 +13,21 @@
 #  limitations under the License.
 
 import struct
-from typing import NamedTuple, Optional, Union
+from typing import TYPE_CHECKING, NamedTuple, Optional, Union
 
+from hathor.conf.get_settings import get_global_settings
 from hathor.transaction import BaseTransaction, Transaction, TxInput
 from hathor.transaction.exceptions import DataIndexError, FinalStackInvalid, InvalidScriptError, OutOfData
 
+if TYPE_CHECKING:
+    from hathor.transaction.scripts.script_context import ScriptContext
+
 
 class ScriptExtras(NamedTuple):
     tx: Transaction
     txin: TxInput
     spent_tx: BaseTransaction
+    input_index: int
```

Comment:
I have changed it to a validated dataclass in https://github.com/HathorNetwork/hathor-core/pull/911/commits/b6112976da77aa63fb70f2eb22b1616ea933eecd, so instead of a note, I assert that an invalid instance cannot be created.

Reply:
Great! Thanks!